solved solution hw6

JlusS opened the pull request:
1) Created corresponding classes (`Square`, `Rectangle`, `RightTriangle`, `Circle`, `IsoscelesTrapezoid`)

2) In the `main()` method created an array of figures 3) Created classes FigureSupplier and ColorSupplier thaat create figures

JlusS committed:
1) Created corresponding classes (`Square`, `Rectangle`, `RightTriangle`, `Circle`, `IsoscelesTrapezoid`)
2) In the `main()` method created an array of figures
3) Created classes FigureSupplier and ColorSupplier thaat create figures

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/HelloWorld.java:
```diff
@@ -1,8 +1,24 @@
+        for (int i = 0; i < figuresArr.length / 2; i++) {
+            figuresArr[i] = figureSupplier.getRandomFigure();
+            System.out.println("Figure: " + figuresArr[i].getClass().getSimpleName() + ", area: "
+                    + figuresArr[i].area() + " sq. units, " + figuresArr[i].getSize() + "color: "
+                        + figuresArr[i].color().toLowerCase());
```
The output string concatenates the size and color fields without a space: `...getSize() + "color: "...`. If the task requires a space between the size and the word 'color', you should add a space after `getSize()`. Please check your task description or checklist for the required output format.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/HelloWorld.java:
```diff
@@ -1,8 +1,24 @@
+        for (int i = 3; i < figuresArr.length; i++) {
+            figuresArr[i] = figureSupplier.getDefaultFigure();
+            System.out.println("Figure: " + figuresArr[i].getClass().getSimpleName() + ", area: "
+                    + figuresArr[i].area() + " sq. units, " + figuresArr[i].getSize() + "color: "
+                        + figuresArr[i].color().toLowerCase());
```
Similar to the previous loop, the output string here does not have a space between the size and color fields. If the output format requires a space, please add it after `getSize()`.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/figures/Circle.java:
```diff
@@ -0,0 +1,34 @@
+    private double radius;
+    private final String color;
+
+    public Circle(int radius, String color) {
+        setRadius(radius);
```
The constructor and setRadius method accept an int for radius, but the field is a double. If the task requires support for double values for radius, you should change the parameter types to double to avoid loss of precision.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/figures/Circle.java:
```diff
@@ -0,0 +1,34 @@
+        return radius;
+    }
+
+    public void setRadius(int radius) {
+        this.radius = radius;
```
The setRadius method accepts an int, but the radius field is a double. If you need to support double values for radius, change the parameter type to double.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/suppliers/Colors.java:
```diff
@@ -0,0 +1,11 @@
+package core.basesyntax.suppliers;
+
+enum Colors {
+    RED,
+    BLACK,
+    WHITE,
+    GREEN,
+    YELLOW,
+    BLUE,
+    ORANGE
+}
```
The Colors enum is package-private. If you need to access this enum from other packages, such as from core.basesyntax.figures, you should declare it as public: `public enum Colors { ... }`. Please verify the access requirements in your task description or checklist.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/suppliers/FigureSupplier.java:
```diff
@@ -0,0 +1,41 @@
+        Random randomSize = new Random();
+
+        if (figures.equals(Figures.Circle)) {
+            return new Circle(randomSize.nextInt(10), colorGeneration());
```
randomSize.nextInt(10) can return 0, which may not be a valid size for a figure (e.g., a circle with radius 0). If the task requires strictly positive sizes, use randomSize.nextInt(9) + 1 to generate values from 1 to 9.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/suppliers/FigureSupplier.java:
```diff
@@ -0,0 +1,41 @@
+        if (figures.equals(Figures.Circle)) {
+            return new Circle(randomSize.nextInt(10), colorGeneration());
+        } else if (figures.equals(Figures.IsoscelesTrapezoid)) {
+            return new IsoscelesTrapezoid(randomSize.nextInt(10), randomSize.nextInt(10),
+                    randomSize.nextInt(10), colorGeneration());
```
randomSize.nextInt(10) can return 0 for sideUp, sideDown, or height. If the task requires strictly positive values, use randomSize.nextInt(9) + 1.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/suppliers/FigureSupplier.java:
```diff
@@ -0,0 +1,41 @@
+            return new IsoscelesTrapezoid(randomSize.nextInt(10), randomSize.nextInt(10),
+                    randomSize.nextInt(10), colorGeneration());
+        } else if (figures.equals(Figures.Rectangle)) {
+            return new Rectangle(randomSize.nextInt(10),
+                    randomSize.nextInt(10), colorGeneration());
```
randomSize.nextInt(10) can return 0 for length or width. If the task requires strictly positive values, use randomSize.nextInt(9) + 1.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/suppliers/FigureSupplier.java:
```diff
@@ -0,0 +1,41 @@
+            return new Rectangle(randomSize.nextInt(10),
+                    randomSize.nextInt(10), colorGeneration());
+        } else if (figures.equals(Figures.RightTriangle)) {
+            return new RightTriangle(randomSize.nextInt(10),
+                    randomSize.nextInt(10), colorGeneration());
```
randomSize.nextInt(10) can return 0 for firstLeg or secondLeg. If the task requires strictly positive values, use randomSize.nextInt(9) + 1.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/suppliers/FigureSupplier.java:
```diff
@@ -0,0 +1,41 @@
+            return new RightTriangle(randomSize.nextInt(10),
+                    randomSize.nextInt(10), colorGeneration());
+        } else {
+            return new Square(randomSize.nextInt(10), colorGeneration());
```
randomSize.nextInt(10) can return 0 for side. If the task requires strictly positive values, use randomSize.nextInt(9) + 1.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/suppliers/Figures.java:
```diff
@@ -0,0 +1,9 @@
+package core.basesyntax.suppliers;
+
+enum Figures {
+    Square,
+    Rectangle,
+    RightTriangle,
+    Circle,
+    IsoscelesTrapezoid
+}
```
The Figures enum is package-private. If you need to access this enum from other packages, such as from core.basesyntax.figures, you should declare it as public: `public enum Figures { ... }`. Please verify the access requirements in your task description or checklist.

mateacademy-ai-mentor reviewed (changes requested):
### Review Summary

Thank you for your submission! There are a few critical issues that need to be addressed before your code can be approved:

1. **Enum Visibility:** Both the `Colors` and `Figures` enums are currently package-private. If these enums are accessed from other packages (such as `core.basesyntax.figures`), they must be declared as `public`. Please update their declarations to `public enum Colors { ... }` and `public enum Figures { ... }` to ensure proper accessibility.

2. **Random Size Generation:** The use of `random.nextInt(10)` for generating figure sizes can result in zero values, which is not valid for figure dimensions (e.g., a circle with radius 0). The task requires all sizes to be strictly positive. Please change all such occurrences to `random.nextInt(9) + 1` so that sizes are always in the range 1 to 9.

3. **Output Formatting:** In your output, the size and color fields are concatenated without a space (e.g., `...getSize() + "color: "...`). If the required output format specifies a space between the size and the word "color", please add it to match the expected format.

4. **Type Consistency in Circle:** The `Circle` class has a `radius` field of type `double`, but the constructor and setter accept `int`. If you need to support double values for the radius, please update the parameter types to `double` to avoid loss of precision.

Please address these issues and resubmit your code. You're very close—just a few important corrections and you'll be all set! 😊 Keep up the good work and pay close attention to the task requirements and checklist details. If you need more information on Java access modifiers or random number generation, feel free to check the [Java documentation](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html) and [Random class](https://docs.oracle.com/javase/8/docs/api/java/util/Random.html). Good luck!

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

> [!IMPORTANT]
> Fix the issues, commit the changes, and then re-request my review using the **`Re-request review` button**[^2] in the Reviewers tab.

[^2]: [Re-request AI review example](https://snipboard.io/nOXxHN.jpg)

> [!TIP]
> If you have any comments about this review, please leave a message in the [Add a comment](#new_comment_field) section below and **mention me with the `@mateacademy-ai-mentor` tag**[^3].
> I will read your message during my next review of your pull request. Please do not reply directly to my comments, as that feature is not supported yet.

[^3]: [Add AI review comments example](https://snipboard.io/hVlOey.jpg)

> [!CAUTION]
> Do not close the pull request until the review is complete. Closing it early can lead to unexpected behavior and disrupt the review process.

JlusS committed:
- Enum Visibility: Both the Colors and Figures enums are currently public
- Random Size Generation: changed all such occurrences to random.nextInt(9) + 1 so that sizes are always in the range 1 to 9.
- Output Formatting

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/suppliers/ColorSupplier.java:
```diff
@@ -0,0 +1,12 @@
+package core.basesyntax.suppliers;
+
+import java.util.Random;
```
Issue: The Colors enum is used in this class, but it is not imported. Please add an import statement for the Colors enum (e.g., `import core.basesyntax.suppliers.Colors;`) to avoid a compilation error.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/suppliers/FigureSupplier.java:
```diff
@@ -0,0 +1,41 @@
+import core.basesyntax.figures.Rectangle;
+import core.basesyntax.figures.RightTriangle;
+import core.basesyntax.figures.Square;
+import java.util.Random;
```
Issue: The Figures enum is used in this class, but it is not imported. Please add an import statement for the Figures enum (e.g., `import core.basesyntax.suppliers.Figures;`) to avoid a compilation error.

mateacademy-ai-mentor reviewed (changes requested):
### Review Summary

Thank you for your hard work! Your implementation meets almost all of the core requirements and your code is well-structured and clear. However, there are two critical issues that need to be addressed before your Pull Request can be approved: the `Colors` enum is used in `ColorSupplier.java` and the `Figures` enum is used in `FigureSupplier.java`, but neither of them are imported. This will cause compilation errors and prevent your code from running as expected. Please add the necessary import statements for these enums in their respective files. Once you fix these issues, your solution will be ready for approval! Keep up the great work! 🚀

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

> [!IMPORTANT]
> Fix the issues, commit the changes, and then re-request my review using the **`Re-request review` button**[^2] in the Reviewers tab.

[^2]: [Re-request AI review example](https://snipboard.io/nOXxHN.jpg)

> [!TIP]
> If you have any comments about this review, please leave a message in the [Add a comment](#new_comment_field) section below and **mention me with the `@mateacademy-ai-mentor` tag**[^3].
> I will read your message during my next review of your pull request. Please do not reply directly to my comments, as that feature is not supported yet.

[^3]: [Add AI review comments example](https://snipboard.io/hVlOey.jpg)

> [!CAUTION]
> Do not close the pull request until the review is complete. Closing it early can lead to unexpected behavior and disrupt the review process.

JlusS committed:
added enum imports to supplier classes

JlusS committed:
removed unused imports

mateacademy-ai-mentor reviewed (commented):
Sorry, the pull request review was unsuccessful for the following reasons:

> [!WARNING]
> **No changes were made in the pull request since the last review.** Please make sure that you have added files to the pull request and committed the changes.
> <sub>*[CODE: 7]*</sub>


If the issue is within your control, please address it.
The next review attempt <ins>will automatically launch</ins> in 1 hour.
If you believe this is an error or need assistance, please contact the support team.

mateacademy-ai-mentor reviewed (commented):
Sorry, the pull request review was unsuccessful for the following reasons:

> [!WARNING]
> **No changes were made in the pull request since the last review.** Please make sure that you have added files to the pull request and committed the changes.
> <sub>*[CODE: 7]*</sub>


If the issue is within your control, please address it.
The next review attempt <ins>will automatically launch</ins> in 1 hour.
If you believe this is an error or need assistance, please contact the support team.

mateacademy-ai-mentor reviewed (commented):
Sorry, the pull request review was unsuccessful for the following reasons:

> [!WARNING]
> **No changes were made in the pull request since the last review.** Please make sure that you have added files to the pull request and committed the changes.
> <sub>*[CODE: 7]*</sub>


If the issue is within your control, please address it.
The next review attempt <ins>will automatically launch</ins> in 1 hour.
If you believe this is an error or need assistance, please contact the support team.

mateacademy-ai-mentor reviewed (commented):
Sorry, the pull request review was unsuccessful for the following reasons:

> [!WARNING]
> **No changes were made in the pull request since the last review.** Please make sure that you have added files to the pull request and committed the changes.
> <sub>*[CODE: 7]*</sub>


If the issue is within your control, please address it.
The next review attempt <ins>will automatically launch</ins> in 1 hour.
If you believe this is an error or need assistance, please contact the support team.

JlusS committed:
removed unused imports

JlusS committed:
added new structure

JlusS committed:
added new structure

liliia-ponomarenko commented on src/main/java/core/basesyntax/enums/Colors.java:
```diff
@@ -0,0 +1,11 @@
+package core.basesyntax.enums;
+
+public enum Colors {
```
```suggestion
public enum Color {
```

liliia-ponomarenko commented on src/main/java/core/basesyntax/Figure.java:
```diff
@@ -0,0 +1,9 @@
+public interface Figure {
+    String getSize();
+
+    double area();
```
The figure should be an abstract class

liliia-ponomarenko commented on src/main/java/core/basesyntax/Figure.java:
```diff
@@ -0,0 +1,9 @@
+
+    double area();
+
+    String color();
```
color it is characteristic  not behavior

liliia-ponomarenko commented on src/main/java/core/basesyntax/HelloWorld.java:
```diff
@@ -1,8 +1,23 @@
 
+        for (int i = 0; i < figuresArr.length / 2; i++) {
+            figuresArr[i] = figureSupplier.getRandomFigure();
+            System.out.println("Figure: " + figuresArr[i].getClass().getSimpleName() + ", area: "
+                    + figuresArr[i].area() + " sq. units, " + figuresArr[i].getSize() + " color: "
+                        + figuresArr[i].color().toLowerCase());
```
each figure should have its own realization

liliia-ponomarenko commented on src/main/java/core/basesyntax/suppliers/FigureSupplier.java:
```diff
@@ -0,0 +1,42 @@
+        Figures figures = Figures.values()[figureIndex];
+        Random randomSize = new Random();
+
+        if (figures.equals(Figures.Circle)) {
+            return new Circle(randomSize.nextInt(9) + 1, colorGeneration());
+        } else if (figures.equals(Figures.IsoscelesTrapezoid)) {
+            return new IsoscelesTrapezoid(randomSize.nextInt(9) + 1, randomSize.nextInt(9) + 1,
```
Use a switch case. Use constant for magic numbers and values

liliia-ponomarenko commented on src/main/java/core/basesyntax/suppliers/ColorSupplier.java:
```diff
@@ -0,0 +1,13 @@
+        int randomIndex = new Random().nextInt(Colors.values().length);
+        return Colors.values()[randomIndex].name();
+    }
+
+}
+
```
remove redundant empty lines

liliia-ponomarenko reviewed (changes requested):
Good start! Let’s  refactor your solution ;)

JlusS committed:
added new interfaces

JlusS committed:
fixed clean-code issues

Elena-Bruyako commented on src/main/java/core/basesyntax/Figure.java:
```diff
@@ -0,0 +1,11 @@
+import core.basesyntax.interfaces.Drawable;
+
+public abstract class Figure implements Drawable, AreaCounter {
+    public abstract String draw();
+
+    public abstract double area();
+
```
```suggestion
    private String color;

    public Figure(String color) {
        this.color = color;
    }
```

Elena-Bruyako commented on src/main/java/core/basesyntax/HelloWorld.java:
```diff
@@ -1,8 +1,19 @@
+
 public class HelloWorld {
+    public static void main(String[] args) {
+        Figure[] figuresArr = new Figure[6];
```
6 is a magic number, create constant

Elena-Bruyako commented on src/main/java/core/basesyntax/HelloWorld.java:
```diff
@@ -1,8 +1,19 @@
+        Figure[] figuresArr = new Figure[6];
+        FigureSupplier figureSupplier = new FigureSupplier();
 
+        for (int i = 0; i < figuresArr.length / 2; i++) {
```
use one for loop to implement main method

Elena-Bruyako commented on src/main/java/core/basesyntax/figures/Circle.java:
```diff
@@ -0,0 +1,25 @@
+    private final double radius;
+    private final Color color;
+
+    public Circle(double radius, Color color) {
```
```suggestion
    private final String color;

    public Circle(double radius, String color) {
```

Elena-Bruyako commented on src/main/java/core/basesyntax/figures/IsoscelesTrapezoid.java:
```diff
@@ -0,0 +1,30 @@
+
+public class IsoscelesTrapezoid extends Figure {
+    private final int sideUp;
+    private final int sideDown;
+    private final int height;
+    private final Color color;
```
same for all figures

JlusS replied:
Can you describe pls what am I have to do. Circle have a radius, triangle have "legs" and square have sides. Do I have to change all names on "legs"?

Elena-Bruyako commented on src/main/java/core/basesyntax/interfaces/Drawable.java:
```diff
@@ -0,0 +1,5 @@
+package core.basesyntax.interfaces;
+
+public interface Drawable {
+    String draw();
```
```suggestion
    void draw();
```

Elena-Bruyako commented on src/main/java/core/basesyntax/suppliers/ColorSupplier.java:
```diff
@@ -0,0 +1,11 @@
+import java.util.Random;
+
+public class ColorSupplier {
+    public Color getRandomColor() {
```
```suggestion
public class ColorSupplier {

    public String getRandomColor() {
```

Elena-Bruyako commented on src/main/java/core/basesyntax/suppliers/FigureSupplier.java:
```diff
@@ -0,0 +1,39 @@
+    private static final int MAX_SIZE = 9;
+
+    public Figure getRandomFigure() {
+        ColorSupplier colorSupplier = new ColorSupplier();
```
make it class-level variable

Elena-Bruyako commented on src/main/java/core/basesyntax/suppliers/FigureSupplier.java:
```diff
@@ -0,0 +1,39 @@
+        ColorSupplier colorSupplier = new ColorSupplier();
+        int figureIndex = new Random().nextInt(Figures.values().length);
+        Figures figures = Figures.values()[figureIndex];
+        Random randomSize = new Random();
```
make it class-level variable

Elena-Bruyako commented on src/main/java/core/basesyntax/suppliers/ColorSupplier.java:
```diff
@@ -0,0 +1,11 @@
+
+public class ColorSupplier {
+    public Color getRandomColor() {
+        return Color.values()[new Random().nextInt(Color.values().length)];
```
make Random class-level variable

Elena-Bruyako commented on src/main/java/core/basesyntax/suppliers/FigureSupplier.java:
```diff
@@ -0,0 +1,39 @@
+        Color color = colorSupplier.getRandomColor();
+
+        return switch (figures) {
+            case Circle -> new Circle(randomSize.nextInt(MAX_SIZE) + 1, color);
```
let's create separate method for r`andomSize.nextInt(MAX_SIZE) + 1` and use it

Elena-Bruyako commented on src/main/java/core/basesyntax/suppliers/FigureSupplier.java:
```diff
@@ -0,0 +1,39 @@
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle(10, Color.WHITE);
```
10 is a magic number, create constant

JlusS committed:
fixed all problems

JJJazl commented on src/main/java/core/basesyntax/enums/Figures.java:
```diff
@@ -0,0 +1,9 @@
+    Rectangle,
+    RightTriangle,
+    Circle,
+    IsoscelesTrapezoid
```
Use upper case

JJJazl commented on src/main/java/core/basesyntax/enums/Figures.java:
```diff
@@ -0,0 +1,9 @@
+package core.basesyntax.enums;
+
+public enum Figures {
```
```suggestion
public enum FigureType {
```

JJJazl commented on src/main/java/core/basesyntax/figures/Circle.java:
```diff
@@ -0,0 +1,24 @@
+
+    @Override
+    public double area() {
+        return 3.14 * (radius * radius);
```
Move it to the contant

JJJazl commented on src/main/java/core/basesyntax/figures/Circle.java:
```diff
@@ -0,0 +1,24 @@
+    public void draw() {
+        System.out.println("Figure: Circle" + ", area: " + area() + " sq. units"
+                + ", radius: " + radius + " units, " + "color: " + color);
+    }
```
Use `String.format()` method. You can also use `FigureType.CIRCLE.name()` instead of "Circle". Fix everywhere

JJJazl commented on src/main/java/core/basesyntax/suppliers/ColorSupplier.java:
```diff
@@ -0,0 +1,13 @@
+    private final Random random = new Random();
+
+    public Color getRandomColor() {
+        return Color.values()[random.nextInt(Color.values().length)];
```
You can move `Color.values().length` to the constant

JJJazl commented on src/main/java/core/basesyntax/suppliers/FigureSupplier.java:
```diff
@@ -0,0 +1,45 @@
+    private static final int DEFAULT_CIRCLE_RADIUS = 10;
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final Random random = new Random();
+    private int figureIndex;
```
```suggestion
```

JJJazl commented on src/main/java/core/basesyntax/suppliers/FigureSupplier.java:
```diff
@@ -0,0 +1,45 @@
+    }
+
+    public Figure getRandomFigure() {
+        figureIndex = random.nextInt(Figures.values().length);
```
Move `Figures.values()` to the constant and reuse
```suggestion
        figureIndex = random.nextInt(FIGURE_ARRAY.length);
```

JJJazl commented on src/main/java/core/basesyntax/suppliers/FigureSupplier.java:
```diff
@@ -0,0 +1,45 @@
+        Color color = colorSupplier.getRandomColor();
+
+        return switch (figures) {
+            case Circle -> new Circle(getRandomSize(), color.name().toLowerCase());
```
Move color.name().toLowerCase() to the variable
```suggestion
            case Circle -> new Circle(getRandomSize(), colorName);
```

JJJazl reviewed (changes requested):
Good! Let's improve your solution)

JlusS committed:
fixed all issues

Elena-Bruyako commented on src/main/java/core/basesyntax/Figure.java:
```diff
@@ -0,0 +1,11 @@
+import core.basesyntax.interfaces.Drawable;
+
+public abstract class Figure implements Drawable, AreaCounter {
+    public abstract void draw();
+
+    public abstract double area();
+
```
you don't need to create abstract method here because you have `Drawable, AreaCounter` for this purpose
`Figure` should have `color`
```suggestion
    private String color;
    
    public Figure(String color) {
        this.color = color;
    }
```

Elena-Bruyako commented on src/main/java/core/basesyntax/figures/Circle.java:
```diff
@@ -0,0 +1,27 @@
+    private final String color;
+
+    public Circle(double radius, String color) {
+        this.radius = radius;
+        this.color = color;
```
for all other figures also
```suggestion
        super(color);
        this.radius = radius;
```

Elena-Bruyako commented on src/main/java/core/basesyntax/figures/Circle.java:
```diff
@@ -0,0 +1,27 @@
+public class Circle extends Figure {
+    private static final double PI = 3.14;
+    private final double radius;
+    private final String color;
```
for all other figures also
```suggestion
```

Elena-Bruyako commented on src/main/java/core/basesyntax/suppliers/ColorSupplier.java:
```diff
@@ -0,0 +1,15 @@
+
+public class ColorSupplier {
+    private static final int COLORSIZE = Color.values().length;
+
```
remove redundant empty line

Elena-Bruyako commented on src/main/java/core/basesyntax/suppliers/ColorSupplier.java:
```diff
@@ -0,0 +1,15 @@
+
+    private final Random random = new Random();
+
+    public Color getRandomColor() {
```
```suggestion
    public String getRandomColor() {
```

Elena-Bruyako commented on src/main/java/core/basesyntax/suppliers/FigureSupplier.java:
```diff
@@ -0,0 +1,47 @@
+    private static final FigureType[] FIGURE_ARRAY = FigureType.values();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final Random random = new Random();
+
```
remove redundant empty line

Elena-Bruyako commented on src/main/java/core/basesyntax/suppliers/FigureSupplier.java:
```diff
@@ -0,0 +1,47 @@
+
+    private FigureType figuresType;
+
+    private int getRandomSize() {
```
private method should be placed after all public method

Elena-Bruyako commented on src/main/java/core/basesyntax/suppliers/FigureSupplier.java:
```diff
@@ -0,0 +1,47 @@
+    public Figure getRandomFigure() {
+        int figureIndex = random.nextInt(FIGURE_ARRAY.length);
+        figuresType = FIGURE_ARRAY[figureIndex];
+        Color color = colorSupplier.getRandomColor();
+        String colorName = color.name().toLowerCase();
```
return `color.name().toLowerCase()` from `getRandomColor()` method
```suggestion
        String color = colorSupplier.getRandomColor();
```

JlusS committed:
fixed all issues
